ls1intum/Artemis
# `Lectures`: Add support for new file extensions in lecture unit file

### Checklist
#### General


#### Server
#### Client

### Motivation and Context

Currently, it is only possible to upload the attachments units in following format: ".png", ".jpg", ".jpeg", ".svg", ".pdf", ".zip"
Uploading e.g. of .docx files is not possible.

The corresponding issue is #5249.
### Description

Now lecture unit file supports additionally following formats: ".png", ".jpg", ".jpeg", ".svg", ".pdf", ".zip", ".doc", ".docx", ".csv", ".xls", ".xlsx", ".ppt", ".pptx", ".pages", ".rtf", ".numbers", ".key", ".odt", ".ods", ".odp", ".odg", ".odf".

### Steps for Testing

Prerequisites:
- 1 Instructor 

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new lecture or open an existing one
4. Scroll down and create a new Unit of type file
5. Click the button "Select file" and ensure that all file formats are supported which are listed on the UI
6. After selecting the file, click submit and ensure that the file unit was created successfully 

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots

![image](https://user-images.githubusercontent.com/10652693/177612064-53782ed5-76fb-4829-8522-3ca403277e65.png)
![image](https://user-images.githubusercontent.com/10652693/177612204-62b20b9f-520c-475e-8a9e-6ee298523b19.png)
![image](https://user-images.githubusercontent.com/10652693/177153160-9789fbdb-6022-47ab-96f9-43e59f1a469a.png)
